aten/src/ATen/core/DistributionsHelper.h

```diff
@@ -144,41 +203,29 @@ struct normal_distribution {
   template <typename RNG>
   C10_HOST_DEVICE inline dist_acctype<T> operator()(RNG generator){
     dist_acctype<T> ret;
-#if !defined(__CUDACC__) && !defined(__HIPCC__)
     // return cached values if available
-    if (std::is_same<T, double>::value) {
-      if (generator->next_double_normal_sample()) {
-        ret = *(generator->next_double_normal_sample()) * stdv + mean;
-        // reset c10::optional to null
-        generator->set_next_double_normal_sample(c10::optional<double>());
-        return ret;
-      }
+    if (std::is_same<T, double>::value && maybe_get_next_double_normal_sample(generator, &ret)) {
+      ret = transformation::normal(ret, mean, stdv);
+      return ret;
+    } else if (maybe_get_next_float_normal_sample(generator, &ret)) {
```

Comment:
This looks wrong. T is double I should never try to get the next float sample, right?

Reply:
addressed